app/src/main/java/com/example/alpaki/presentation/fragments/RegisterViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.alpaki.presentation.fragments
+
+import androidx.hilt.Assisted
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.*
+import com.example.domain.models.UserCandidate
+import com.example.domain.usecases.Register
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+
+class RegisterViewModel @ViewModelInject constructor(
+    @Assisted private val savedStateHandle: SavedStateHandle,
+    private val register: Register
+) : ViewModel(), LifecycleObserver {
```

Comment:
Nie trzeba implementować LifecycleObserver, viewModelScope powinien być dostępny po dodaniu odp. dependencji liffecycle do gradle'a i tyle :)

Reply:
Chyba, że to nie do tego 😄